### ISSUE TITLE
Invisible NoCSG collision brushes don't have collision

### ISSUE THREAD

**Issue by Henry00IS** [sabresaurus/SabreCSG#106]
Apparently the NoCSG brushes with collision enabled and visibility off, are just ignored for collision completely. While it's very rare that you'd ever need to do this (e.g. long build times, severe splitting from cylinders), this combination shouldn't behave unexpectedly. It's probably some if-sentence that's wrong somewhere.

![image](https://user-images.githubusercontent.com/7905726/39661916-4fb97a56-505a-11e8-883a-fc676b0c1545.png)

**Comment by Kerfuffles** [comment]
Looking into this, it appears as though it's only accounting for BuiltVisualPolygons when calculating collision, in the [CoreBuild()](https://github.com/sabresaurus/SabreCSG/blob/3c51a4e2f500f2257cdf2ec8fe54914f16aead38/Scripts/Core/BuildEngine/CSGFactory.cs#L108) method in [CSGFactory]( SabreCSG/Scripts/Core/BuildEngine/CSGFactory.cs ), starting at line [402](https://github.com/sabresaurus/SabreCSG/blob/3c51a4e2f500f2257cdf2ec8fe54914f16aead38/Scripts/Core/BuildEngine/CSGFactory.cs#L402).

I'm not sure how to go about fixing this, but it's a place to start though.

**Comment by Henry00IS** [comment]
Yes this turns out to be a lot more complicated and deeply rooted than I initially thought. Fixing this the wrong way will cause NoCSG brushes to have CSG operations.
